clientlibs/js/src/functions/addMetrics.ts

```diff
@@ -8,25 +8,23 @@ export default async function addMetrics(
   clientSessionId: string,
   metrics: (ISingleMetric | IGroupMetric)[]
 ): Promise<Interfaces.IMetric[]> {
-  try {
-    const response = await fetchDataService(
-      url,
-      token,
-      clientSessionId,
-      { metricUnit: metrics },
-      Types.REQUEST_TYPES.POST
-    );
-    if (response.status) {
-      response.data = response.data.map((metric) => {
+  const response = await fetchDataService(
+    url,
+    token,
+    clientSessionId,
+    { metricUnit: metrics },
+    Types.REQUEST_TYPES.POST
+  );
+  if (response.status) {
```

Comment:
Can you check the status to be in range of 200 – 299. And every where else. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

Reply:
Sorry I haven't seen the fetchDataService. It is already checking that ignore the above comment.